core/jobs/batch_jobs/state_interaction_validation_jobs.py

```diff
@@ -0,0 +1,1382 @@
+                # Validates link tag attributes exists.
+                elif rte_component['id'] == 'oppia-noninteractive-link':
+                    if (
+                        'text-with-value' not in
+                        rte_component['customization_args'] or
+                        rte_component['customization_args'][
+                            'text-with-value'] == 'Not found'
+                    ):
+                        rte_components_errors.append(
+                            f'State - {key} having RTE link tag do not '
+                            f'have text-with-value attribute'
+                        )
+
+                    if (
+                        'url-with-value' not in
+                        rte_component['customization_args'] or
+                        rte_component['customization_args'][
+                            'url-with-value'] == 'Not found'
+                    ):
+                        rte_components_errors.append(
+                            f'State - {key} having RTE link tag do not '
+                            f'have url-with-value attribute'
+                        )
+
+            if len(rte_components_errors) > 0:
+                states_with_values.append(
+                    {
+                        'state_name': key,
+                        'rte_components_errors': rte_components_errors
+                    }
+                )
+        return states_with_values
+
+    @staticmethod
+    def _set_lower_and_upper_bounds(
+        range_var, lower_bound, upper_bound,
+        lb_inclusive, ub_inclusive
+    ) -> None:
+        """Sets the lower and upper bounds for the range_var, mainly
+        we need to set the range so to keep track if any other rule's
+        range lies in between or not to prevent redundancy
+
+        Args:
+            range_var: dict[str, Any]. To keep track of each rule's
+                ans group index, rule spec index, lower bound, upper bound,
+                lb inclusive, ub inclusive.
+
+            lower_bound: int. The lower bound.
+
+            upper_bound: int. The upper bound.
+
+            lb_inclusive: bool. If lower bound is inclusive.
+
+            ub_inclusive: bool. If upper bound is inclusive.
+        """
+        range_var['lower_bound'] = lower_bound
+        range_var['upper_bound'] = upper_bound
+        range_var['lb_inclusive'] = lb_inclusive
+        range_var['ub_inclusive'] = ub_inclusive
+
+    @staticmethod
+    def _is_enclosed_by(range_var, range_ele) -> bool:
+        """Checks whether the ranges of rules enclosed or not
+
+        Args:
+            range_var: dict[str, Any]. To keep track of each rule's
+                ans group index, rule spec index, lower bound, upper bound,
+                lb inclusive, ub inclusive.
+
+            range_ele: dict[str, Any]. To keep track of other rule's
+                ans group index, rule spec index, lower bound, upper bound,
+                lb inclusive, ub inclusive.
```

Comment:
```suggestion
            range_var: dict[str, Any]. To keep track of each rule's
                ans group index, rule spec index, lower bound, upper bound,
                lb inclusive, ub inclusive.
            range_ele: dict[str, Any]. To keep track of other rule's
                ans group index, rule spec index, lower bound, upper bound,
                lb inclusive, ub inclusive.
```

The descriptions are exactly the same for both of these, from that it is hard to understand how they differ, please explain that somewhere in the docstring.

Reply:
So basically, the `range_var` is the variable that we have to compare and `range_ele` is the one to which we have to compare.
I will add one line mentioning that.